tests/unit/s2n_self_talk_session_resumption_test.c

```diff
@@ -26,13 +26,23 @@
     (((client->handshake.handshake_type) & HELLO_RETRY_REQUEST) \
      && ((server->handshake.handshake_type) & HELLO_RETRY_REQUEST))
 
+#define EXPECT_TICKETS_SENT(conn, count) EXPECT_OK(s2n_assert_tickets_sent(conn, count))
+
 struct s2n_early_data_test_case {
     bool ticket_supported;
     bool client_supported;
     bool server_supported;
     bool expect_success;
 };
 
+static S2N_RESULT s2n_assert_tickets_sent(struct s2n_connection *conn, uint16_t expected_tickets_sent)
+{
+    uint16_t tickets_sent = 0;
+    RESULT_GUARD_POSIX(s2n_connection_get_tickets_sent(conn, &tickets_sent));
+    RESULT_ENSURE_EQ(tickets_sent, expected_tickets_sent);
+    return S2N_RESULT_OK;
+}
```

Comment:
Is there a downside to making the assertions in the function? That way you don't even need the macro at all

```suggestion
static void s2n_assert_tickets_sent(struct s2n_connection *conn, uint16_t expected_tickets_sent)
{
    uint16_t tickets_sent = 0;
    EXPECT_SUCCESS(s2n_connection_get_tickets_sent(conn, &tickets_sent));
    EXPECT_EQ(tickets_sent, expected_tickets_sent);
}
```

Reply:
Unfortunately there is. The EXPECT macros report the line that failed. So if the EXPECT macros are on line 41/42, then whatever test actually fails will just report a failure on line 41/42. That makes it difficult to determine which test failed.

Reply:
Opened an issue to improve this: https://github.com/aws/s2n-tls/issues/3057